src/exportJarSteps/ExportJarTaskProvider.ts

```diff
@@ -57,13 +135,173 @@ class ExportJarTaskTerminal implements Pseudoterminal {
+            for (const project of projectList) {
+                const runtimeClassPaths: string[] = [];
+                await this.getClasspaths(project.uri, "runtime", runtimeClassPaths, runtimeDependencies);
+                classPathMap.set(project.name, runtimeClassPaths);
+                const testClassPaths: string[] = [];
+                await this.getClasspaths(project.uri, "test", testClassPaths, testDependencies);
+                testClassPathMap.set(project.name, testClassPaths);
+            }
+            this.stepMetadata.classpaths = await this.resolveClassPaths(runtimeDependencies, testDependencies, classPathMap, testClassPathMap);
+        }
         await createJarFile(this.stepMetadata);
         this.closeEmitter.fire();
     }
 
     public close(): void {
 
     }
+
+    private toAbsolute(path: string): string {
+        if (path.length > 0 && path[0] === "!") {
+            const positivePath = path.substring(1);
+            if (!isAbsolute(positivePath)) {
+                return "!" + join(this.stepMetadata.workspaceFolder.uri.fsPath, positivePath);
+            }
+        } else if (!isAbsolute(path)) {
+            return join(this.stepMetadata.workspaceFolder.uri.fsPath, path);
+        }
+        return path;
+    }
+
+    private async getClasspaths(projectUri: string, _scope: string, classpaths: string[], dependencies: string[]): Promise<void> {
+        const extension: Extension<any> | undefined = extensions.getExtension("redhat.java");
+        const extensionApi: any = await extension?.activate();
+        const classpathResult: IClasspathResult = await extensionApi.getClasspaths(projectUri, { scope: _scope });
+        for (const classpath of classpathResult.classpaths) {
+            if (extname(classpath) === ".jar") {
+                dependencies.push(classpath);
+            } else {
+                classpaths.push(classpath);
+            }
+        }
+        for (const classpath of classpathResult.modulepaths) {
+            if (extname(classpath) === ".jar") {
+                dependencies.push(classpath);
+            } else {
+                classpaths.push(classpath);
+            }
+        }
+    }
+
+    private async resolveClassPaths(runtimeDependencies: string[], testDependencies: string[],
+                                    classPathMap: Map<string, string[]>, testClassPathMap: Map<string, string[]>): Promise<IClassPath[]> {
+        // tslint:disable-next-line: no-invalid-template-strings
+        const regExp: RegExp = new RegExp("\\${(.*)}");
+        const classPathArray: string[] = [];
+        const dependencies: string[] = [];
+        for (const element of this.stepMetadata.elements) {
+            const variableResult = element.match(regExp);
+            if (_.isEmpty(variableResult) || variableResult.length <= 1) {
+                classPathArray.push(upath.normalizeSafe(this.toAbsolute(element)));
```

Comment:
Any reason not using `path.normalize`?

Reply:
Glob pattern needs a posix path as its input. In windows, simply running `path.normalize()` or `path.posix.normalize()` can't translate the path. Instead of using `upath`, I will write a new utility to translate windows path to posix path.

Reply:
I see. It's ok to use `upath` then.